### PR TITLE
Add on_change for expandable

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #615

```python
from src.taipy.gui import Gui

var = False
v = "This is an expandableeeeeee"

def on_action(state):
    state.v = "Expandable text changed"
    state.var = True

def on_change(state, var, val, module):
    print(var, val, module)

def custom_on_change(state, var, val, module):
    print("custom on change", var, val)

Gui("""
# Test page

<|This is the label|button|on_action=on_action|>

<|{v}|expandable|expanded={var}|on_change=custom_on_change|

This is the content

|>

""").run(run_browser=False)
```

### CODE REVIEW

File: src/taipy/gui/renderers/builder.py [hunk 1]
```diff
@@ -795,7 +817,10 @@ def set_attributes(self, attributes: t.List[tuple]):  # noqa: C901
                     self.set_boolean_attribute(attr[0], val)
             elif var_type == PropertyType.dynamic_boolean:
                 self.__set_dynamic_bool_attribute(
-                    attr[0], _get_tuple_val(attr, 2, False), _get_tuple_val(attr, 3, False)
+                    attr[0],
+                    _get_tuple_val(attr, 2, False),
+                    _get_tuple_val(attr, 3, False),
+                    _get_tuple_val(attr, 4, True),
```
Comment (dinhlongviolin1) [inline src/taipy/gui/renderers/builder.py:823]: Main addition in builder.py

**Review by FredLL-Avaiga — APPROVED** [review]
Well done

File: gui/src/components/Taipy/Expandable.tsx [hunk 2]
```diff
@@ -33,8 +35,11 @@ interface ExpandableProps extends TaipyActiveProps {
 }
 
 const Expandable = (props: ExpandableProps) => {
-    const { id, expanded = true, defaultExpanded, title, defaultTitle, page, partial } = props;
-    const [opened, setOpened] = useState(defaultExpanded === undefined ? expanded : defaultExpanded);
+    const { id, expanded, defaultExpanded, title, defaultTitle, page, partial, updateVars, propagate = true } = props;
+    const { dispatch } = useContext(TaipyContext);
```
Comment (FredLL-Avaiga) [inline gui/src/components/Taipy/Expandable.tsx]: could have use: const dispatch = useDispatch();

**Review by FredLL-Avaiga — APPROVED** [review]

File: gui/src/components/Taipy/Expandable.tsx (hunk 2 above)
Comment (dinhlongviolin1) [reply gui/src/components/Taipy/Expandable.tsx]: thanks!

File: src/taipy/gui/renderers/factory.py [hunk 3]
```diff
@@ -159,8 +159,9 @@ class _Factory:
             [
                 ("id",),
                 ("page",),
-                ("expanded", PropertyType.dynamic_boolean, True),
+                ("expanded", PropertyType.dynamic_boolean, True, True, False),
```
Comment (FabienLelaquais) [inline src/taipy/gui/renderers/factory.py:162]: 👌

**Review by FabienLelaquais — APPROVED** [review]
Looks just fine!

**Pull request merged**